### PR TITLE
Add get collection and get self

### PR DESCRIPTION
**Pull request by agouriou**
Add getCollection has a shortcut to perform a get request returning collections.
Add getSelf to perform a get request on self link.

See #61 for previous discussions.

### CODE REVIEW

**Comment by agouriou** [comment]
Build failed on "npm install" fetching "expand-tilde-1.2.2"

**Comment by maennchen** [comment]
I restarted the build, I hope it runs through.

File: src/resource/hal-resource-client.factory.js [hunk 1]
```diff
@@ -201,6 +223,20 @@ export default function HalResourceClientFactory($q, $injector, $halConfiguratio
...
+
+    /**
+     * Execute a HTTP GET request on self
+     *
+     * @param {Object|null} urlParams
+     * @param {Object}      options
+     * @return {Promise}
+     */
+    function $getSelf(urlParams, options) {
+      return $http(angular.extend({}, options, {
```
Comment (maennchen) [inline src/resource/hal-resource-client.factory.js]: Why don't you just use `$request('GET', 'self', null, options)` to do the request?

I don't think a self link shouldn't have `urlParams` by design. What do you think?

**Comment by agouriou** [comment]
$request returns the current resource if it's called with self. 
See in hal-resource-client line 61
`if(method === 'GET' &&
         rel === $halConfiguration.selfLink) {
        return $q.resolve(resource);
      }
`

Regarding `urlParams` on self link, I tend to agree with you, but I'm not sure if's HATEOAS good practice or not...

**Comment by maennchen** [comment]
You're right. Let's extract the http request from `$request` and use it in both places.

I think that parameters in self links are not allowed by design. I'd omit that parameter.

**Pull request merged**